pulpcore/pulpcore/app/viewsets/repository.py

```diff
@@ -178,6 +187,63 @@ def destroy(self, request, repository_pk, number):
         )
         return OperationPostponedResponse([async_result], request)
 
+    def validate_content_url(self, url):
+        """
+        Validates that the path belongs to a Content resource
+
+        Args:
+            url (str): href that needs validation
+
+        Returns:
+            PK of the Content
+
+        Raises:
+            :class:`rest_framework.exceptions.ValidationError`: When the URL is not a valid
+                Content resource
+        """
+
+        try:
+            resolver = resolve(urlparse(url).path)
+        except Resolver404 as e:
+            raise ValidationError(detail=_("Invalid hyperlink. %s") % url)
+
+        if not resolver.view_name.startswith('content') or not resolver.view_name.endswith(
+                'detail'):
+            # URL exists, but does not correspond to Content resource
+            raise ValidationError(detail=_("Invalid hyperlink. %s") % url)
+
+        content_field = serializers.HyperlinkedRelatedField(view_name=resolver.view_name,
+                                                            queryset=Content.objects.all(),
+                                                            source='*', initial=url)
+        try:
+            content_field.run_validation(data=url)
+        except ValidationError as e:
+            # Append the URL of missing Content to the error message
+            e.detail[0] = "%s %s" % (e.detail[0], url)
+            raise e
+        return resolver.kwargs['pk']
+
+    def create(self, request, repository_pk):
+        add_content_units = []
+        remove_content_units = []
+        if 'add_content_units' in request.data:
+            for url in request.data['add_content_units']:
+                add_content_units.append(self.validate_content_url(url))
+
+        if 'remove_content_units' in request.data:
+            for url in request.data['remove_content_units']:
+                remove_content_units.append(self.validate_content_url(url))
```

Comment:
It seems weird that you're appending the result of a validation function to a list of content units. Probably should call the function something like `parse_content_url` instead since it parses out the pk and validating the url is a secondary purpose.

Reply:
You are right. Originally I was just passing the whole URL to the task and then I switched to using just the PKs. I'll rename the method.